FSW/src/util/comUtil.cpp

```diff
@@ -40,28 +40,49 @@ int command[commandQueueSize];
  */
 void commandHandling()
 {
-    int newCommand = getCommand();
-    newCommand
-}
+    int newCommandCode = getCommand();
+    
+    if (newCommandCode != -1) // if the new command is valid, add it to the queue
+    {
+        queueCommand(newCommandCode);
+    }
 
+}
 
 
-byte getCommand();
+int getCommand();
 {
     // read Serial
     if (Serial.available() > 0) // check if there is serial input
     {
-        int serialInput = Serial.parseInt();
-        return serialInput;
+        int serialInput = Serial.parseInt(); // read the command
+        if ((serialInput <= static_cast<int>(Command::DoNothing)) && (serialInput >= 0)) // check if command is valid
```

Comment:
I like this logic, however would it potentially be more useful to split up these prompts? Right now the same error is printed regardless of if the check fails due to the command code being larger than `DoNothing` or if it fails due to a negative `serialInput`

Something like "Command failed due to invalid serial input" separate from "Command code outside enum bounds"

Reply:
Good call. Changed to print out the value of the invalid command and the range of acceptable commands.